### DISCUSSION TITLE
Can file as the ICs?

### DISCUSSION THREAD

**Discussion by KangChenRui** [idaholab/moose#20883] [Q&A General] [answered]
Can I use a file as an initial condition in MOOSE?

**Comment by GiudGiud** [comment, marked as answer]
hello

what kind of file? CSV? exodus?

Guillaume

**Reply by KangChenRui** [reply]
Hi, Guillaume

Any files supported by moose will do, I don't know how to use them as there is no such example for the ICs module.
Can you show an example in the input file?

Thanks for your reply!

Chen Rui

**Reply by GiudGiud** [reply]
There's a couple options for loading from CSV. I m working on a doc page
For data to interpolate on the mesh:
https://mooseframework.inl.gov/source/functions/PiecewiseMultilinear.html
https://mooseframework.inl.gov/source/functions/PiecewiseMulticonstant.html
For data to load directly on nodes/elements/blocks:
https://mooseframework.inl.gov/source/functions/PiecewiseConstantFromCSV.html
combined with
https://mooseframework.inl.gov/source/userobject/PropertyReadFile.html

There are input files attached to the documentation.

**Reply by KangChenRui** [reply]
Thank you very much for your guidance.

I will study it.

Chen Rui
